### PR TITLE
Omit rendering resources when not supplied

### LINKED ISSUE #10350: Empty resource block in generated quick-install.yaml
It seems this was introduced by #10296 . @maxbischoff can you take a look?

The quick-install.yaml is generated by running `make -C install/kubernetes clean all`

### PR DESCRIPTION
**Pull request by maxbischoff**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Fixes: #10350

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10363)

### CODE REVIEW

File: install/kubernetes/quick-install.yaml [hunk 1]
```diff
@@ -613,10 +611,9 @@ spec:
         - mountPath: /tmp/cilium/config-map
           name: cilium-config-path
           readOnly: true
-        resources:
-          {}
       hostNetwork: true
       restartPolicy: Always
+      priorityClassName: system-cluster-critical
```
Comment (aanm) [inline install/kubernetes/quick-install.yaml]: Please remove this one manually for now

**Review by aanm — CHANGES_REQUESTED** [review]
Thanks for taking a look at this

File: install/kubernetes/cilium/charts/agent/templates/daemonset.yaml [hunk 2]
```diff
@@ -290,8 +294,10 @@ spec:
 {{- end }}
         - mountPath: /var/run/cilium
           name: cilium-run
+{{- if .Values.initResources }}
```
Comment (maxbischoff) [inline install/kubernetes/cilium/charts/agent/templates/daemonset.yaml:297]: This allows explicitly setting `initResources` to an empty dict to cause the resources to not be rendered.
I can also leave this change out if you prefer.

Comment (aanm) [reply install/kubernetes/cilium/charts/agent/templates/daemonset.yaml:297]: > This allows explicitly setting `initResources` to an empty dict to cause the resources to not be rendered.
> I can also leave this change out if you prefer.

how could this happen exactly? It's not clear to me.

Comment (maxbischoff) [reply install/kubernetes/cilium/charts/agent/templates/daemonset.yaml:297]: A user could supply a values.yaml to the agent chart with following content:

```
initResources: {}
```

File: install/kubernetes/quick-install.yaml (hunk 1 above)
Comment (maxbischoff) [reply install/kubernetes/quick-install.yaml]: removed.

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by maxbischoff** [comment]
Seems like a Jenkins issue:
```
16:36:49  ERROR: Error fetching remote repo 'origin'
16:36:49  hudson.plugins.git.GitException: Failed to fetch from https://github.com/cilium/cilium
```

**Comment by aanm** [comment]
test-me-please

**Pull request merged**